docs/contributing.rst

```diff
@@ -0,0 +1,181 @@
+environment, and install the developer dependencies.
+
+::
+
+    $ git clone git@github.com:HyperionGray/trio-websocket.git
+    $ cd trio-websocket
+    $ python3 -m venv venv
+    $ source venv/bin/activate
+    (venv) $ pip install --editable .[dev]
+
+
+Unit Tests
+----------
+
+.. note::
+
+    This project has unit tests that are configured to run on all pull requests
+    to automatically check for regressions. Each pull request should include
+    unit test coverage before it is merged.
+
+The unit tests are written with `the PyTest framework
+<https://docs.pytest.org/en/latest/>`__. You can quickly run all unit tests from
+the project's root with a simple command::
+
+    (venv) $ pytest
+    === test session starts ===
+    platform linux -- Python 3.6.6, pytest-3.8.0, py-1.6.0, pluggy-0.7.1
+    rootdir: /home/johndoe/code/trio-websocket, inifile: pytest.ini
+    plugins: trio-0.5.0, cov-2.6.0
+    collected 27 items
+
+    tests/test_connection.py ........................... [100%]
+
+    === 27 passed in 0.41 seconds ===
+
+You can enable code coverage reporting by adding the ``-cov=trio_websocket``
+option to PyTest::
+
+    (venv) $ pytest --cov=trio_websocket
+    === test session starts ===
+    platform linux -- Python 3.6.6, pytest-3.8.0, py-1.6.0, pluggy-0.7.1
+    rootdir: /home/johndoe/code/trio-websocket, inifile: pytest.ini
+    plugins: trio-0.5.0, cov-2.6.0
+    collected 27 items
+
+    tests/test_connection.py ........................... [100%]
+
+    ---------- coverage: platform darwin, python 3.7.0-final-0 -----------
+    Name                         Stmts   Miss  Cover
+    ------------------------------------------------
+    trio_websocket/__init__.py     369     33    91%
+    trio_websocket/version.py        1      0   100%
+    ------------------------------------------------
+    TOTAL                          370     33    91%
+
+
+    === 27 passed in 0.57 seconds ===
+
+Documentation
+-------------
+
+This documentation is stored in the repository in the ``/docs/`` directory. It
+is written with `RestructuredText markup
+<http://docutils.sourceforge.net/rst.html>`__ and processed by `Sphinx
+<http://www.sphinx-doc.org/en/stable/>`__. To build documentation, go into the
+documentation directory and run this command::
+
+    $ make html
+
+The finished documentation can be found in ``/docs/_build/``. It is published
+automatically to `Read The Docs <https://readthedocs.org/>`__.
+
+Autobahn Client Tests
+---------------------
+
+The Autobahn Test Suite contains over 500 integration tests for WebSocket
+servers and clients. These test suites are contained in a `Docker
+<https://www.docker.com/>`__ container. You will need to install Docker before
+you can run these integration tests.
+
+To test the client, you will need two terminal windows. In the first terminal,
+run the following commands::
+
+    $ cd autobahn
+    $ docker run -it --rm \
```

Comment:
perhaps make makefile targets for autobahn server and client

Reply:
I like the Makefile idea, too. I'll tackle that in #61 as well.